File: #1422.maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        
        int size = s.size();
        map<char, int>m1;
        for (int i = 0; i < size; i++) {
            m1[s[i]]++;
        }
        
        map<char, int>m2;
        int result = 0;
        for (int i = 0; i < size - 1; i++) {
            m1[s[i]]--;
            m2[s[i]]++;
            result = std::max(result, m1['1'] + m2['0']);
        }
        return result;
    }
};

File: #1422.maximum-score-after-splitting-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var maxScore = function(s) {
    let zero = 0;
    let one = [...s.matchAll(/1/g)].length;
    let res = 0;
    for (let i = 0; i + 1 < s.length; i++) {
        s[i] === "0" ? zero++ : one--;
        res = Math.max(zero + one, res);
    }
    return res;
};
